pkg/actor/partitioned_update.go

```diff
@@ -0,0 +1,216 @@
+
+	clientset, err := kubernetes.NewForConfig(up.config)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create kubernetes clientset")
+	}
+
+	// test to see if we are running inside of Kubernetes
+	// If we are running inside of k8s we will not find this file.
+	runningInsideK8s := inK8s("/var/run/secrets/kubernetes.io/serviceaccount/token")
+
+	serviceName := cluster.PublicServiceName()
+	if runningInsideK8s {
+		log.Info("operator is running inside of kubernetes, connecting to service for db connection")
+	} else {
+		serviceName = fmt.Sprintf("%s-0.%s.%s", cluster.Name(), cluster.Name(), cluster.Namespace())
+		log.Info("operator is NOT inside of kubernetes, connnecting to pod ordinal zero for db connection")
+	}
+
+	// The connection needs to use the discovery service name because of the
+	// hostnames in the SSL certificates
+	conn := &database.DBConnection{
+		Ctx:              ctx,
+		Client:           up.client,
+		RestConfig:       up.config,
+		ServiceName:      serviceName,
+		Namespace:        cluster.Namespace(),
+		DatabaseName:     "system", // TODO we need to use variable instead of string
+		Port:             cluster.Spec().GRPCPort,
+		RunningInsideK8s: runningInsideK8s,
+	}
+
+	// see https://github.com/cockroachdb/cockroach-operator/issues/204 for above TODO
+
+	if cluster.Spec().TLSEnabled {
+		conn.UseSSL = true
+		conn.ClientCertificateSecretName = cluster.ClientTLSSecretName()
+		conn.RootCertificateSecretName = cluster.NodeTLSSecretName()
+	}
+
+	// TODO we may have an error case where the operator will not finish an update, but will
+	// still try to make a database connection.
+	// see https://github.com/cockroachdb/cockroach-operator/issues/205
+
+	// Create a new database connection for the update.
+	// TODO we may want to create this db connection later
+	// see https://github.com/cockroachdb/cockroach-operator/issues/207
+
+	db, err := database.NewDbConnection(conn)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create database connection")
+	}
+	log.Info("opened db connection")
+	defer db.Close()
+
+	// TODO test downgrades
+	// see https://github.com/cockroachdb/cockroach-operator/issues/208
+
+	err = update.UpdateClusterCockroachVersion(
+		ctx,
+		specStr,
+		clientset,
+		wantContainer,
+		specVer,
+		podUpdateTimeout,
+		podMaxPollingInterval,
+		db,
+		sleeper,
+		stsName,
+		cluster.Namespace(),
+		log,
+	)
+
+	// TODO set status so that we will not try to update the cluster again
+	// TODO set status to rollback cluster?
+	// This work is pending the status field updates
+	// see https://github.com/cockroachdb/cockroach-operator/issues/209
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to update sts with partitioned update: %s", stsName)
+	}
+
+	// TODO set status that we are completed.
+	log.Info("update completed with partitioned update", "new version", specStr)
+	CancelLoop(ctx)
+	return nil
+}
+
+// inK8s checks to see if the a file exists
+func inK8s(file string) bool {
```

Comment:
Should file just be a constant that's used by this method?

Reply:
So I wanted to test it, so I moved it to a param, and also it is only used in a single line of code, so I hardcoded it.  Do we want to make it a const?